app/src/forms/proxy/routes.js

```diff
@@ -0,0 +1,21 @@
+const cors = require('cors');
+
+const { currentUser } = require('../auth/middleware/userAccess');
+const controller = require('./controller');
+
+const routes = require('express').Router();
+
+// need to allow cors for OPTIONS call
+// formio component will call OPTIONS pre-flight
+routes.options('/external', cors());
+
+// called with encrypted headers, no current user!!!
+routes.get('/external', cors(), async (_req, res, next) => {
+  await controller.callExternalApi(_req, res, next);
```

Comment:
This is the part I'm not getting - why use the bearer token to create this encrypted object, and then use the encrypted object in this call, and extract what was in the bearer token? Why not use the bearer token here?

Reply:
This call comes from the form component (like drop down control) which doesn't have access to the token unless we make it available to the form designer. If we do that, then we have no control over what a form designer uses that token for...

So the initial call on load we do have the token because CHEFS frontend is making the call (just like fetching the form schema). Only the CHEFS backend can decrypt this and read the true token. Not sure if we will even leave the token in that encrypted bundle or not - but it is up to the designer whether they want it added to the call to their API.

Reply:
Thanks for the info! I completely forgot that the form runs in a sort-of sandbox. This does bring up another problem I need to sort out, and maybe that solution will be useful - will see! (and in the meantime I'll try to understand the rest)